zerver/lib/mention.py

```diff
@@ -465,3 +469,25 @@ def get_user_group_mention_display_name(user_group: NamedUserGroup) -> StrPromis
         return SystemGroups.GROUP_DISPLAY_NAME_MAP[user_group.name]
 
     return user_group.name
+
+
+def sender_can_mention_group(sender: UserProfile | None, named_group: NamedUserGroup) -> bool:
+    can_mention_group = named_group.can_mention_group
+
+    if (
+        hasattr(can_mention_group, "named_user_group")
+        and can_mention_group.named_user_group.name == SystemGroups.EVERYONE
+    ):
+        return True
+
+    assert sender is not None
+
+    if is_cross_realm_bot_email(sender.delivery_email):
+        return False
+
+    return user_has_permission_for_group_setting(
+        can_mention_group.id,
+        sender,
+        NamedUserGroup.GROUP_PERMISSION_SETTINGS["can_mention_group"],
+        direct_member_only=False,
+    )
```

Comment:
This extraction does seem helpful, though. So maybe make just extracting this the second commit.